Mathlib/Algebra/GroupWithZero/WithZero.lean

```diff
@@ -407,15 +407,26 @@ end WithZero
 
 namespace MonoidWithZeroHom
 
-protected lemma map_eq_zero_iff {G₀ G₀' : Type*} [GroupWithZero G₀]
-    [MulZeroOneClass G₀'] [Nontrivial G₀']
-    {f : G₀ →*₀ G₀'} {x : G₀} :
+variable {G₀' : Type*} [GroupWithZero G₀']
+
+/-- The map `withZeroUnitsEquiv` as a `MonoidWithZeroHom`. -/
+def withZeroUnitsHom [DecidablePred (fun a : G₀' ↦ a = 0)] :
+    WithZero G₀'ˣ →*₀ G₀' where
+  __ := WithZero.withZeroUnitsEquiv
+  map_zero' := rfl
+  map_one' := rfl
+
+lemma withZeroUnitsHom_inj [DecidablePred (fun a : G₀' ↦ a = 0)] :
+    Injective (withZeroUnitsHom (G₀' := G₀')) := MulEquiv.injective WithZero.withZeroUnitsEquiv
+
+protected lemma map_eq_zero_iff {G₀ : Type*} [MulZeroOneClass G₀] [Nontrivial G₀]
```

Comment:
```suggestion
protected lemma map_eq_zero_iff {M₀ : Type*} [MulZeroOneClass M₀] [Nontrivial M₀]
```
to avoid using group-like variable names for monoid-like variables